src/pygnssutils/gnssntripclient.py

```diff
@@ -486,83 +499,131 @@ def _read_thread(
-                            datatype,
-                            stopevent,
-                            ggainterval,
-                            output,
-                        )
-                    elif rc == "1":  # retrieved sourcetable
-                        stopevent.set()
-                        self._connected = False
-                        self._app_update_status(False)
-                    else:  # error message
-                        stopevent.set()
-                        self._connected = False
-                        self._app_update_status(False, (f"Error!: {rc}", "red"))
-        except (
-            socket.gaierror,
-            ConnectionRefusedError,
-            ConnectionAbortedError,
-            ConnectionResetError,
-            BrokenPipeError,
-            TimeoutError,
-            OverflowError,
-        ):
-            stopevent.set()
-            self._connected = False
-        except ssl.SSLCertVerificationError as err:
-            tip = (
-                f" - try using '{server[4:]}' rather than '{server}' for the NTRIP caster URL"
-                if "certificate is not valid for 'www." in err.strerror
-                else (
-                    f" - try adding the NTRIP caster URL SSL certificate to {findcacerts()}"
-                    if "unable to get local issuer certificate" in err.strerror
-                    else ""
                 )
-            )
-            print(f"SSL Certificate Verification Error{tip}\n{err}")
-            stopevent.set()
-            self._connected = False
-            self._app_update_status(False, (f"Error!: {err.strerror[0:32]}", "red"))
+                print(f"SSL Certificate Verification Error{tip}\n{err}")
+                self._retrycount = self._retries
+                stopevent.set()
+                self._connected = False
+                self._app_update_status(False, (f"Error!: {err.strerror[0:60]}", "red"))
+
+            except (
+                BrokenPipeError,
+                ConnectionAbortedError,
+                ConnectionRefusedError,
+                ConnectionResetError,
+                OverflowError,
+                socket.gaierror,
+                ssl.SSLError,
+                TimeoutError,
+            ) as err:
+                errm = str(repr(err))
+                erra = f"Connection Error {errm.split('(', 1)[0]}!"
+                errl = f"Error connecting to {server}:{port}/{mountpoint}: {errm}"
+                if self._retrycount == self._retries:
+                    stopevent.set()
+                    self._connected = False
+                else:
+                    self._retrycount += 1
+                    errr = (
+                        f". Retrying in {self._retryinterval * self._retrycount} secs "
+                        f"({self._retrycount}/{self._retries}) ..."
+                    )
+                    erra += errr
+                    errl += errr
+                self._do_log(errl)

```

Comment:
Sorry to chime in @semuadmin, this is an excellent enhancement to increase the robustness of the ntrip server! 

Just suggestion, can we lower the log level here to VERBOSITY_LOW for better trouble shooting ? 
This line here set the log level to medium which is the same as the rtcm data in _do_write(). Imagine if we are in a network instable scenario, the rtcm logs will very likely to overwrite the timeout and retry messages here. 

Perhaps we should increase the rtcm log message  level to VERBOSITY_HIGH  in _do_write() ?

Reply:
Agreed (and will check other error messages for similar omission).

Have provisionally amended do_write() to log the entire parsed message if verbosity is HIGH, but only the message identity if verbosity is MEDIUM e.g.

```shell
2024-07-24 08:16:30.483616: Streaming RTCM data from euref-ip.net:443/DARE00GBR0 ...
2024-07-24 08:16:30.489642: 1097
2024-07-24 08:16:30.521891: 1087
2024-07-24 08:16:30.523049: 1077
2024-07-24 08:16:30.523199: 1019
2024-07-24 08:16:30.523328: 1042
2024-07-24 08:16:30.523451: 1046
2024-07-24 08:16:30.523571: 1020
^C2024-07-24 08:16:31.163890: Streaming terminated
```